### ISSUE TITLE
--multi-user cmd arg doesn't work properly, different output dirs for different sessions

### ISSUE THREAD

**Issue by ashish-aesthisia** [comfyanonymous/ComfyUI#3417]
### Issue
I want to keep different directories for users accessing the single endpoint session for comfyUI. 

### How to Reproduce
After launching Comfy for the first time I saw this message in the console:
`****** For multi-user setups add the --multi-user CLI argument to enable multiple user profiles. ******`
So I restarted it with this extra argument and It seems that Comfy already has a basic user system. 

It seems to be a pretty basic, file-based user system used only for settings and templates (third screenshot), but It may help us to add some authenticated user sessions.

![image (28)](https://github.com/comfyanonymous/ComfyUI/assets/108272701/7f9dc4e8-4f74-4d57-9c88-16af17111575)

#### Commands

`python3 main.py --multi-user`

Then when i launch session, I get to created a user

![image (29)](https://github.com/comfyanonymous/ComfyUI/assets/108272701/a2e36d2b-d78c-45fb-ac75-a88ae42b4398)

But when i generate an image, output supposed to go within the user folder as shown in screenshot. I switched to another user but the outputs are all saved in the same folder. This users stuff seems to affect only settings.

### Tested on 
- Mac M1 Pro & Air
- Linux machine with Nvidia enabled GPU. 

### Expectation

User folders should save output to each correspondence user. 

Correct me if i misunderstood the use of --multi-user arg. 

### Ask
As a team we are using single GPU enabled machine to run all of our sessions. We want to separate the output based on different browser sessions. 
Is there anyway to make it possible?

**Comment by a67793581** [comment]
This is an unimplemented feature, it is not clear if the author has any plans to add it, if there is a lack of developers, I can help implement this feature.

**Comment by Tobe2d** [comment]
Any update on this?

**Comment by DmitryKripakov** [comment]
I'm also really looking forward to this feature :)

**Comment by Rouzax** [comment]
Is there any way to set the Output dir per user profile?
I know that I can redirect the output with `--output-directory` but it would be nice if I could set that to the User profile

**Comment by julien-moment** [comment]
Same here, I'm looking for a solution to nest input/output folders in a user folder since I have multiple users working with the same Comfy server.

+1 to implement a feature to implement that 🙏

**Comment by Lalimec** [comment]
I am also trying to use a single comfy installation from an efs mounted to different instances. Is there any progress on this of i will just try to use separate start scripts for each user.

**Comment by Zeeshanzt** [comment]
HI, any progress on this feature?

**Comment by bigcat88** [comment]
@comfyanonymous are you ready to accept PR regarding isolating workflow results for each user?

Because we have the same problem as in this issue, we need `user1` not to see `user2` results, both incoming and outgoing files in Comfy UI.

I have two options, either duplicate this code and just add support for it to our project, or work a little harder and add this PR to Comfy right away.

**Comment by oumad** [comment]
This --multi-user arg sounds good, but apparently it doesn't provide privacy for different users, it's more like a profiles option as opposed to users. Meaning any other "user" could still select yours from the drop down list and open comfyui as you. It's not like gradio's user authentication. If we can at least have the latter, we can ourselves implement the necessary code to separate the output/input directories based on the currently connected user. But if ComfyUI can already support that, it would be great !

**Comment by hbardak** [comment]
I agree that this multi-user environment should ensure privacy for input and output folders. Is there any update on this front?
